jsonnet/kube-prometheus/kube-prometheus-anti-affinity.libsonnet

```diff
@@ -1,23 +1,22 @@
-local k = import 'github.com/ksonnet/ksonnet-lib/ksonnet.beta.4/k.libsonnet';
-local statefulSet = k.apps.v1.statefulSet;
-local affinity = statefulSet.mixin.spec.template.spec.affinity.podAntiAffinity.preferredDuringSchedulingIgnoredDuringExecutionType;
-local matchExpression = affinity.mixin.podAffinityTerm.labelSelector.matchExpressionsType;
-
 {
   local antiaffinity(key, values, namespace) = {
     affinity: {
       podAntiAffinity: {
         preferredDuringSchedulingIgnoredDuringExecution: [
-          affinity.new() +
-          affinity.withWeight(100) +
-          affinity.mixin.podAffinityTerm.withNamespaces(namespace) +
-          affinity.mixin.podAffinityTerm.withTopologyKey('kubernetes.io/hostname') +
-          affinity.mixin.podAffinityTerm.labelSelector.withMatchExpressions([
-            matchExpression.new() +
-            matchExpression.withKey(key) +
-            matchExpression.withOperator('In') +
-            matchExpression.withValues(values),
-          ]),
+          {
+            podAffinityTerm: {
+              namespaces: [namespace],
+              topologyKey: 'kubernetes.io/hostname',
+              labelSelector: {
+                matchExpressions: [{
+                  key: key,
+                  operator: 'In',
+                  values: values,
+                }],
+              },
+              weight: 100,
```

Comment:
this will break whole kube-prometheus currently
error validating "/tmp/generated336709272/jsonnet/templates/jsonnet/extra/monitoring/alertmanager-alertmanager.yaml": error validating data: [ValidationError(Alertmanager.spec.affinity.podAntiAffinity.preferredDuringSchedulingIgnoredDuringExecution[0].podAffinityTerm): unknown field "weight" in com.coreos.monitoring.v1.Alertmanager.spec.affinity.podAntiAffinity.preferredDuringSchedulingIgnoredDuringExecution.podAffinityTerm, ValidationError(Alertmanager.spec.affinity.podAntiAffinity.preferredDuringSchedulingIgnoredDuringExecution[0]): missing required field "weight" in com.coreos.monitoring.v1.Alertmanager.spec.affinity.podAntiAffinity.preferredDuringSchedulingIgnoredDuringExecution]; if you choose to ignore these errors, turn validation off with --validate=false

the indentation is incorrect in `weight` parameter.

Reply:
Thanks for reporting this. I'll take a look and provide an fix.